packages/ra-core/src/dataProvider/useDelete.ts

```diff
@@ -318,7 +316,7 @@ export const useDelete = <
         if (mode.current === 'pessimistic') {
             return mutation.mutate(
                 { resource: callTimeResource, ...callTimeParams },
-                { onSuccess, onSettled, onError }
+                callTimeOptions
```

Comment:
you're passing too many options to the underlying mutation (dataProvider.delete()  doesn't understand `mutationMode`)